## Importing Pandas Library

In [1]:
import pandas as pd

## Importing csv file

In [2]:
df = pd.read_csv('(Error Free) sgcarmart.csv')

Our (Error Free) sgcarmart dataset was obtained by web-scrapping the sgcarmart.com website through a web scraper tool from webscraper.io.

In [3]:
df.columns

Index(['RefLink', 'CarName', 'Price', 'Depreciation', 'Mileage', 'Road Tax',
       'Dereg Value', 'COE', 'Engine Cap', 'Curb Weight', 'Type of Vehicle',
       'Reg Date', 'Transmission', 'OMV', 'ARF', 'Power', 'No of Owners'],
      dtype='object')

In [4]:
df.head()

RefLink  \
0  https://www.sgcarmart.com/used_cars/info.php?I...   
1  https://www.sgcarmart.com/used_cars/info.php?I...   
2  https://www.sgcarmart.com/used_cars/info.php?I...   
3  https://www.sgcarmart.com/used_cars/info.php?I...   
4  https://www.sgcarmart.com/used_cars/info.php?I...   

                                             CarName     Price  \
0                              Volvo V40 T4 R-Design  $115,800   
1  Land Rover Range Rover Sport Diesel 3.0A 7-Seater  $188,888   
2                             Volvo XC60 T5 Momentum  $170,000   
3              Toyota Voxy 2.0A X (COE till 09/2028)   $68,888   
4               Audi A3 Sedan 1.4A TFSI CoD S-tronic  $100,800   

                                  Depreciation                 Mileage  \
0  $14,120 /yr\nView models with similar depre    17,000 km (6.3k /yr)   
1  $27,100 /yr\nView models with similar depre  133,000 km (24.6k /yr)   
2  $22,680 /yr\nView models with similar depre   58,000 km (16.5k /yr)   
3                                  $10,580 /yr    176,000 km (13k /yr)   
4  $13,290 /yr\nView models with similar depre   96,775 km (28.8k /yr)   

     Road Tax                    Dereg Value      COE Engine Cap Curb Weight  \
0  $1,176 /yr   $44,075 as of today (change)  $34,000   1,969 cc    1,454 kg   
1  $3,376 /yr  $115,977 as of today (change)  $56,956   2,993 cc    2,183 kg   
2  $1,176 /yr   $54,933 as of today (change)  $32,311   1,969 cc    1,841 kg   
3  $1,674 /yr   $21,727 as of today (change)  $33,377   1,986 cc    1,560 kg   
4    $624 /yr   $40,187 as of today (change)  $32,302   1,395 cc    1,320 kg   

  Type of Vehicle                                   Reg Date Transmission  \
0       Hatchback  08-Jul-2019\n(7yrs 3mths 13days COE left)         Auto   
1             SUV   28-Oct-2016\n(4yrs 7mths 3days COE left)         Auto   
2             SUV  20-Sep-2018\n(6yrs 5mths 26days COE left)         Auto   
3             MPV        25-Sep-2008\n(6yrs 6mths  COE left)         Auto   
4    Luxury Sedan  13-Nov-2018\n(6yrs 7mths 19days COE left)         Auto   

       OMV       ARF               Power  No of Owners  
0  $24,074   $25,704  140.0 kW (187 bhp)             1  
1  $86,985  $128,573  225.0 kW (301 bhp)             2  
2  $38,070   $45,298  187.0 kW (250 bhp)             1  
3  $22,387   $22,387  105.0 kW (140 bhp)             4  
4  $23,548   $24,968  110.0 kW (147 bhp)             1

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10361 entries, 0 to 10360
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   RefLink          10361 non-null  object
 1   CarName          10361 non-null  object
 2   Price            10361 non-null  object
 3   Depreciation     10361 non-null  object
 4   Mileage          10361 non-null  object
 5   Road Tax         10361 non-null  object
 6   Dereg Value      10361 non-null  object
 7   COE              10361 non-null  object
 8   Engine Cap       10361 non-null  object
 9   Curb Weight      10361 non-null  object
 10  Type of Vehicle  10361 non-null  object
 11  Reg Date         10361 non-null  object
 12  Transmission     10361 non-null  object
 13  OMV              10361 non-null  object
 14  ARF              10361 non-null  object
 15  Power            10361 non-null  object
 16  No of Owners     10361 non-null  int64 
dtypes: int64(1), object(16)
memory 

## Cleaning the data:

Since we have webscrapped the information from the 'sgcarmart.com' website, the data collected are disorganised. Thus, there is a need to clean the data before we start our analysis.

### 1) Modifying 'RefLink' column

The 'RefLink' column holds the reference links to each car listing in sgcarmart.  
An example is:

In [6]:
df['RefLink'][0]

'https://www.sgcarmart.com/used_cars/info.php?ID=1081489&DL=2178'

Thus, instead of keeping the reference links as data, we would like to retain the CarID.  
As seen in the above link, the CarID is **1081489**.

In [7]:
array = []

for index in df['RefLink'].index:
    array.append(df['RefLink'].loc[index].split("?")[1])
    
for string in array:
    array[array.index(string)] = string.split("&")[0].lstrip("ID=")

df.insert(1, "CarID", array)

df.head()

RefLink    CarID  \
0  https://www.sgcarmart.com/used_cars/info.php?I...  1081489   
1  https://www.sgcarmart.com/used_cars/info.php?I...  1070886   
2  https://www.sgcarmart.com/used_cars/info.php?I...  1082885   
3  https://www.sgcarmart.com/used_cars/info.php?I...  1080087   
4  https://www.sgcarmart.com/used_cars/info.php?I...  1079487   

                                             CarName     Price  \
0                              Volvo V40 T4 R-Design  $115,800   
1  Land Rover Range Rover Sport Diesel 3.0A 7-Seater  $188,888   
2                             Volvo XC60 T5 Momentum  $170,000   
3              Toyota Voxy 2.0A X (COE till 09/2028)   $68,888   
4               Audi A3 Sedan 1.4A TFSI CoD S-tronic  $100,800   

                                  Depreciation                 Mileage  \
0  $14,120 /yr\nView models with similar depre    17,000 km (6.3k /yr)   
1  $27,100 /yr\nView models with similar depre  133,000 km (24.6k /yr)   
2  $22,680 /yr\nView models with similar depre   58,000 km (16.5k /yr)   
3                                  $10,580 /yr    176,000 km (13k /yr)   
4  $13,290 /yr\nView models with similar depre   96,775 km (28.8k /yr)   

     Road Tax                    Dereg Value      COE Engine Cap Curb Weight  \
0  $1,176 /yr   $44,075 as of today (change)  $34,000   1,969 cc    1,454 kg   
1  $3,376 /yr  $115,977 as of today (change)  $56,956   2,993 cc    2,183 kg   
2  $1,176 /yr   $54,933 as of today (change)  $32,311   1,969 cc    1,841 kg   
3  $1,674 /yr   $21,727 as of today (change)  $33,377   1,986 cc    1,560 kg   
4    $624 /yr   $40,187 as of today (change)  $32,302   1,395 cc    1,320 kg   

  Type of Vehicle                                   Reg Date Transmission  \
0       Hatchback  08-Jul-2019\n(7yrs 3mths 13days COE left)         Auto   
1             SUV   28-Oct-2016\n(4yrs 7mths 3days COE left)         Auto   
2             SUV  20-Sep-2018\n(6yrs 5mths 26days COE left)         Auto   
3             MPV        25-Sep-2008\n(6yrs 6mths  COE left)         Auto   
4    Luxury Sedan  13-Nov-2018\n(6yrs 7mths 19days COE left)         Auto   

       OMV       ARF               Power  No of Owners  
0  $24,074   $25,704  140.0 kW (187 bhp)             1  
1  $86,985  $128,573  225.0 kW (301 bhp)             2  
2  $38,070   $45,298  187.0 kW (250 bhp)             1  
3  $22,387   $22,387  105.0 kW (140 bhp)             4  
4  $23,548   $24,968  110.0 kW (147 bhp)             1

Now we can remove the 'RefLink' column.

In [8]:
df = df.drop(columns=['RefLink'])

### 2)Trim CarName
'CarName' column holds the name of the car in each listing. However, some of the data contains excess.  
For example:

In [9]:
df['CarName'][3]

'Toyota Voxy 2.0A X (COE till 09/2028)'

Thus, we would like to trim off the excess string "(COE till 09/2028)" as the duration of COE is included in the 'Reg Date' column.

In [10]:
array = []

for index in df['CarName'].index:
    array.append(df['CarName'].loc[index].split("(")[0])

for string in array:
    array[array.index(string)] = string.rstrip()

df.insert(2, "UsedCarName", array)

df.head()

CarID                                            CarName  \
0  1081489                              Volvo V40 T4 R-Design   
1  1070886  Land Rover Range Rover Sport Diesel 3.0A 7-Seater   
2  1082885                             Volvo XC60 T5 Momentum   
3  1080087              Toyota Voxy 2.0A X (COE till 09/2028)   
4  1079487               Audi A3 Sedan 1.4A TFSI CoD S-tronic   

                                         UsedCarName     Price  \
0                              Volvo V40 T4 R-Design  $115,800   
1  Land Rover Range Rover Sport Diesel 3.0A 7-Seater  $188,888   
2                             Volvo XC60 T5 Momentum  $170,000   
3                                 Toyota Voxy 2.0A X   $68,888   
4               Audi A3 Sedan 1.4A TFSI CoD S-tronic  $100,800   

                                  Depreciation                 Mileage  \
0  $14,120 /yr\nView models with similar depre    17,000 km (6.3k /yr)   
1  $27,100 /yr\nView models with similar depre  133,000 km (24.6k /yr)   
2  $22,680 /yr\nView models with similar depre   58,000 km (16.5k /yr)   
3                                  $10,580 /yr    176,000 km (13k /yr)   
4  $13,290 /yr\nView models with similar depre   96,775 km (28.8k /yr)   

     Road Tax                    Dereg Value      COE Engine Cap Curb Weight  \
0  $1,176 /yr   $44,075 as of today (change)  $34,000   1,969 cc    1,454 kg   
1  $3,376 /yr  $115,977 as of today (change)  $56,956   2,993 cc    2,183 kg   
2  $1,176 /yr   $54,933 as of today (change)  $32,311   1,969 cc    1,841 kg   
3  $1,674 /yr   $21,727 as of today (change)  $33,377   1,986 cc    1,560 kg   
4    $624 /yr   $40,187 as of today (change)  $32,302   1,395 cc    1,320 kg   

  Type of Vehicle                                   Reg Date Transmission  \
0       Hatchback  08-Jul-2019\n(7yrs 3mths 13days COE left)         Auto   
1             SUV   28-Oct-2016\n(4yrs 7mths 3days COE left)         Auto   
2             SUV  20-Sep-2018\n(6yrs 5mths 26days COE left)         Auto   
3             MPV        25-Sep-2008\n(6yrs 6mths  COE left)         Auto   
4    Luxury Sedan  13-Nov-2018\n(6yrs 7mths 19days COE left)         Auto   

       OMV       ARF               Power  No of Owners  
0  $24,074   $25,704  140.0 kW (187 bhp)             1  
1  $86,985  $128,573  225.0 kW (301 bhp)             2  
2  $38,070   $45,298  187.0 kW (250 bhp)             1  
3  $22,387   $22,387  105.0 kW (140 bhp)             4  
4  $23,548   $24,968  110.0 kW (147 bhp)             1

In [11]:
df = df.drop(columns=['CarName'])

In [12]:
df = df.rename(columns={'UsedCarName' : 'CarName'})

### 3) Remove any empty data

In most columns, they include N.A data. Thus, we would like to remove any data points that contain such data.

In [13]:
array = []

for index in df['Price'].index:
    if not('N.A' in df['Price'].loc[index]):   
        array.append(index)

df = df.loc[array].reset_index(drop=True)
df.shape

(10323, 17)

In [14]:
array = []

for index in df['Depreciation'].index:
    if not('N.A' in df['Depreciation'].loc[index]):   
        array.append(index)

df = df.loc[array].reset_index(drop=True)
df.shape

(10148, 17)

In [15]:
array = []

for index in df['Mileage'].index:
    if not('N.A' in df['Mileage'].loc[index]):   
        array.append(index)

df = df.loc[array].reset_index(drop=True)
df.shape

(8505, 17)

In [16]:
array = []

for index in df['Road Tax'].index:
    if not('N.A' in df['Road Tax'].loc[index]):   
        array.append(index)

df = df.loc[array].reset_index(drop=True)
df.shape

(8504, 17)

In [17]:
array = []

for index in df['Dereg Value'].index:
    if not('N.A' in df['Dereg Value'].loc[index]):   
        array.append(index)

df = df.loc[array].reset_index(drop=True)
df.shape

(8112, 17)

In [18]:
array = []

for index in df['COE'].index:
    if not('N.A' in df['COE'].loc[index]):   
        array.append(index)

df = df.loc[array].reset_index(drop=True)
df.shape

(8103, 17)

In [19]:
array = []

for index in df['Curb Weight'].index:
    if not('N.A' in df['Curb Weight'].loc[index]):   
        array.append(index)

df = df.loc[array].reset_index(drop=True)
df.shape

(8008, 17)

### 4) Convert Columns datatypes

We would like to trim unnecessary data in each column and convert it into a suitable datatype.

In [20]:
df.dtypes

CarID              object
CarName            object
Price              object
Depreciation       object
Mileage            object
Road Tax           object
Dereg Value        object
COE                object
Engine Cap         object
Curb Weight        object
Type of Vehicle    object
Reg Date           object
Transmission       object
OMV                object
ARF                object
Power              object
No of Owners        int64
dtype: object

Changing CarID data type to int type.

In [21]:
df = df.astype({'CarID' : 'int64'})

Modifying data in 'Price' Column & converting to int data type.

In [22]:
for index in df['Price'].index:
    temp = df['Price'].loc[index].lstrip("$")
    temp = temp.replace(',',"")
    df['Price'] = df['Price'].replace(df['Price'].loc[index],temp)
    

In [23]:
df = df.astype({'Price' : 'int64'})

Modifying data in 'Depreciation' Column & converting to int data type.

In [24]:
for index in df['Depreciation'].index:
    temp = df['Depreciation'].loc[index].lstrip("$")
    if ('/yr' in temp):
        temp = temp.split("/")[0]
    temp = temp.rstrip()
    temp = temp.replace(',',"")
    df['Depreciation'] = df['Depreciation'].replace(df['Depreciation'].loc[index],temp)

In [25]:
df = df.astype({'Depreciation' : 'int64'})

Modifying data in 'Mileage' Column & converting to int data type.

In [26]:
for index in df['Mileage'].index:
    temp = df['Mileage'].loc[index].split("km")[0]
    temp = temp.rstrip()
    temp = temp.replace(',',"")
    df['Mileage'] = df['Mileage'].replace(df['Mileage'].loc[index],temp)

In [27]:
df = df.astype({'Mileage' : 'int64'})

Modifying data in 'Road Tax' Column & converting to int data type.

In [28]:
for index in df['Road Tax'].index:
    temp = df['Road Tax'].loc[index].lstrip("$")
    temp = temp.split("/yr")[0]
    temp = temp.rstrip()
    temp = temp.replace(',',"")
    df['Road Tax'] = df['Road Tax'].replace(df['Road Tax'].loc[index],temp)

In [29]:
df = df.astype({'Road Tax' : 'int64'})

Modifying data in 'Dereg Value' Column & converting to int data type.

In [30]:
for index in df['Dereg Value'].index:
    temp = df['Dereg Value'].loc[index].lstrip("$")
    temp = temp.split("as")[0]
    temp = temp.rstrip()
    temp = temp.replace(',',"")
    df['Dereg Value'] = df['Dereg Value'].replace(df['Dereg Value'].loc[index],temp)

In [31]:
df = df.astype({'Dereg Value' : 'int64'})

Modifying data in 'COE' Column & converting to int data type.

In [32]:
for index in df['COE'].index:
    temp = df['COE'].loc[index].lstrip("$")
    temp = temp.replace(',',"")
    df['COE'] = df['COE'].replace(df['COE'].loc[index],temp)

In [33]:
df = df.astype({'COE' : 'int64'})

Modifying data in 'Engine Cap' Column & converting to int data type.

In [34]:
for index in df['Engine Cap'].index:
    temp = df['Engine Cap'].loc[index]
    temp = temp.replace('cc',"")
    temp = temp.rstrip()
    temp = temp.replace(',',"")
    df['Engine Cap'] = df['Engine Cap'].replace(df['Engine Cap'].loc[index],temp)

In [35]:
df = df.astype({'Engine Cap' : 'int64'})

Modifying data in 'Curb Weight' Column & converting to int data type.

In [36]:
for index in df['Curb Weight'].index:
    temp = df['Curb Weight'].loc[index]
    temp = temp.replace('kg',"")
    temp = temp.rstrip()
    temp = temp.replace(',',"")
    df['Curb Weight'] = df['Curb Weight'].replace(df['Curb Weight'].loc[index],temp)

In [37]:
df = df.astype({'Curb Weight' : 'int64'})

Modifying data in 'Reg Date' Column.

Here, we are going to:
1) Convert 'Reg Date' data into datetime64[ns] data type  
2) Extract the duration of COE left

In [38]:
df['Reg Date']

0       08-Jul-2019\n(7yrs 3mths 13days COE left)
1        28-Oct-2016\n(4yrs 7mths 3days COE left)
2       20-Sep-2018\n(6yrs 5mths 26days COE left)
3             25-Sep-2008\n(6yrs 6mths  COE left)
4       13-Nov-2018\n(6yrs 7mths 19days COE left)
                          ...                    
8003    16-Jun-2016\n(4yrs 2mths 22days COE left)
8004     30-Oct-2015\n(3yrs 7mths 5days COE left)
8005    31-Jan-2019\n(6yrs 10mths 6days COE left)
8006      31-Jul-2008\n(6yrs 1mth 6days COE left)
8007    18-Dec-2018\n(6yrs 8mths 23days COE left)
Name: Reg Date, Length: 8008, dtype: object

In [39]:
COE_array = []

for index in df['Reg Date'].index:
    temp_array = df['Reg Date'].loc[index].split("\n")
    date = temp_array[0]
    if (len(temp_array) > 1): 
        COE_left = temp_array[1]
    COE_left = COE_left.lstrip("(")
    COE_left = COE_left.split("COE")[0]
    COE_left = COE_left.rstrip()
    
    COE_array.append(COE_left)
    
    df['Reg Date'] = df['Reg Date'].replace(df['Reg Date'].loc[index],date)
    

In [40]:
df['Reg Date'] = pd.to_datetime(df['Reg Date'])

In [41]:
converted_COE_array = []

for String in COE_array:
    int_array = [0,0,0]
    temp_array = String.split()
    for temp_string in temp_array:
        if 'yr' in temp_string:
            temp = temp_string.split("y")[0]
            int_array[0] = int(temp)
        elif 'mth' in temp_string:
            temp = temp_string.split("m")[0]
            int_array[1] = int(temp)
        elif 'day' in temp_string:
            temp = temp_string.split("d")[0]
            int_array[2] = int(temp)
    converted_COE_array.append(int_array)

In [42]:
df.insert(df.columns.get_loc('Transmission'), "COE Expiry Date", None)

In [43]:
df['COE Expiry Date'] = pd.to_datetime(df['COE Expiry Date'])

In the following code, we set the date to **24 March 2022** as it was the date when we webscrapped the website.

In [44]:
timeseries = []

for index in df['COE Expiry Date'].index:
    ts = pd.to_datetime("2022/3/24")
    ts = ts + pd.DateOffset(years = converted_COE_array[index][0], months = converted_COE_array[index][1], days = converted_COE_array[index][2])
    timeseries.append(ts)
    
df['COE Expiry Date'] = timeseries

In [45]:
df.insert(df.columns.get_loc('Transmission'), "COE Duration", None)

In [46]:
array = []

for index in df['COE Duration'].index:
    ts = pd.to_datetime("2022/3/24")
    difference = (df['COE Expiry Date'].loc[index] - ts).days
    array.append(difference)

df['COE Duration'] = array

In [47]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8008 entries, 0 to 8007
Data columns (total 19 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   CarID            8008 non-null   int64         
 1   CarName          8008 non-null   object        
 2   Price            8008 non-null   int64         
 3   Depreciation     8008 non-null   int64         
 4   Mileage          8008 non-null   int64         
 5   Road Tax         8008 non-null   int64         
 6   Dereg Value      8008 non-null   int64         
 7   COE              8008 non-null   int64         
 8   Engine Cap       8008 non-null   int64         
 9   Curb Weight      8008 non-null   int64         
 10  Type of Vehicle  8008 non-null   object        
 11  Reg Date         8008 non-null   datetime64[ns]
 12  COE Expiry Date  8008 non-null   datetime64[ns]
 13  COE Duration     8008 non-null   int64         
 14  Transmission     8008 non-null   object 

Modifying data in 'OMV' Column & converting to int data type.

In [48]:
for index in df['OMV'].index:
    temp = df['OMV'].loc[index].lstrip("$")
    temp = temp.replace(',',"")
    df['OMV'] = df['OMV'].replace(df['OMV'].loc[index],temp)

In [49]:
df = df.astype({'OMV' : 'int64'})

Modifying data in 'ARF' Column & converting to int data type.

In [50]:
for index in df['ARF'].index:
    temp = df['ARF'].loc[index].lstrip("$")
    temp = temp.replace(',',"")
    df['ARF'] = df['ARF'].replace(df['ARF'].loc[index],temp)

In [51]:
df = df.astype({'ARF' : 'int64'})

Modifying data in 'Power' Column & converting to float data type.  
We are extracting kW instead of bhp.

In [52]:
for index in df['Power'].index:
    temp = df['Power'].loc[index].split("kW")[0]
    temp = temp.rstrip()
    df['Power'] = df['Power'].replace(df['Power'].loc[index],temp)

In [53]:
df = df.astype({'Power' : 'float64'})

### 5) Adding 'CarAge' Column

From 'Reg Date', we can determine the age of the car in days.

In [54]:
df.insert(df.columns.get_loc('Price'), "CarAge", None)

In [55]:
array = []

for index in df['Reg Date'].index:
    ts = pd.to_datetime("2022/3/24")
    difference = (ts - df['Reg Date'].loc[index]).days
    array.append(difference)

df['CarAge'] = array

## Finished Cleaning

Here's how our clean dataset looks:

In [56]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8008 entries, 0 to 8007
Data columns (total 20 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   CarID            8008 non-null   int64         
 1   CarName          8008 non-null   object        
 2   CarAge           8008 non-null   int64         
 3   Price            8008 non-null   int64         
 4   Depreciation     8008 non-null   int64         
 5   Mileage          8008 non-null   int64         
 6   Road Tax         8008 non-null   int64         
 7   Dereg Value      8008 non-null   int64         
 8   COE              8008 non-null   int64         
 9   Engine Cap       8008 non-null   int64         
 10  Curb Weight      8008 non-null   int64         
 11  Type of Vehicle  8008 non-null   object        
 12  Reg Date         8008 non-null   datetime64[ns]
 13  COE Expiry Date  8008 non-null   datetime64[ns]
 14  COE Duration     8008 non-null   int64  

In [57]:
df

CarID                                            CarName  CarAge  \
0     1081489                              Volvo V40 T4 R-Design     990   
1     1070886  Land Rover Range Rover Sport Diesel 3.0A 7-Seater    1973   
2     1082885                             Volvo XC60 T5 Momentum    1281   
3     1080087                                 Toyota Voxy 2.0A X    4928   
4     1079487               Audi A3 Sedan 1.4A TFSI CoD S-tronic    1227   
...       ...                                                ...     ...   
8003  1083139                            BMW 3 Series 320i Sport    2107   
8004  1041762                         Audi Q3 1.4A TFSI S-Tronic    2337   
8005  1071766    Mercedes-Benz E-Class E200 Avantgarde Panoramic    1148   
8006  1076962                            BMW 6 Series 630i Coupe    4984   
8007  1068494                                  MINI One 1.5A 5DR    1192   

       Price  Depreciation  Mileage  Road Tax  Dereg Value    COE  Engine Cap  \
0     115800         14120    17000      1176        44075  34000        1969   
1     188888         27100   133000      3376       115977  56956        2993   
2     170000         22680    58000      1176        54933  32311        1969   
3      68888         10580   176000      1674        21727  33377        1986   
4     100800         13290    96775       624        40187  32302        1395   
...      ...           ...      ...       ...          ...    ...         ...   
8003   94888         18090    28099      1210        48971  55100        1998   
8004   75800         16340    92000       624        43574  60000        1395   
8005  171800         20330    38162      1202        70581  32200        1991   
8006   78800         12900    95000      3335        24965  40881        2996   
8007  103900         14150    34000       684        31566  28000        1499   

      Curb Weight Type of Vehicle   Reg Date COE Expiry Date  COE Duration  \
0            1454       Hatchback 2019-07-08      2029-07-07          2662   
1            2183             SUV 2016-10-28      2026-10-27          1678   
2            1841             SUV 2018-09-20      2028-09-19          2371   
3            1560             MPV 2008-09-25      2028-09-24          2376   
4            1320    Luxury Sedan 2018-11-13      2028-11-12          2425   
...           ...             ...        ...             ...           ...   
8003         1450    Luxury Sedan 2016-06-16      2024-02-19           697   
8004         1480             SUV 2015-10-30      2024-02-19           697   
8005         1615    Luxury Sedan 2019-01-31      2024-02-19           697   
8006         1540      Sports Car 2008-07-31      2028-04-30          2229   
8007         1225       Hatchback 2018-12-18      2028-04-30          2229   

     Transmission    OMV     ARF  Power  No of Owners  
0            Auto  24074   25704  140.0             1  
1            Auto  86985  128573  225.0             2  
2            Auto  38070   45298  187.0             1  
3            Auto  22387   22387  105.0             4  
4            Auto  23548   24968  110.0             1  
...           ...    ...     ...    ...           ...  
8003         Auto  35497   36696  135.0             1  
8004         Auto  29913   33879  110.0             1  
8005         Auto  51474   64654  135.0             1  
8006         Auto  83812   83812  200.0             3  
8007         Auto  24947   16926   75.0             1  

[8008 rows x 20 columns]

## Export to csv

We can export this csv file using the following command:

In [58]:
df.to_csv('Clean sgcarmart dataset as of 24MAR2022.csv', index=False)